Lib/test/test_typing.py

```diff
@@ -155,8 +156,43 @@ def test_cannot_instantiate(self):
             type(NoReturn)()
 
 
-class TypeVarTests(BaseTestCase):
+class SelfTests(BaseTestCase):
+    def test_basics(self):
+        class Foo:
+            def bar(self) -> Self: ...
+
+        self.assertEqual(gth(Foo.bar), {'return': Self})
+
+    def test_repr(self):
+        self.assertEqual(repr(Self), 'typing.Self')
+
+    def test_cannot_subscript(self):
+        with self.assertRaises(TypeError):
+            Self[int]
+
+    def test_cannot_subclass(self):
+        with self.assertRaises(TypeError):
+            class C(type(Self)):
+                pass
+
+    def test_cannot_init(self):
+        with self.assertRaises(TypeError):
+            Self()
+        with self.assertRaises(TypeError):
+            type(Self)()
+
+    def test_no_isinstance(self):
+        with self.assertRaises(TypeError):
+            isinstance(1, Self)
+        with self.assertRaises(TypeError):
+            issubclass(int, Self)
 
+    def test_alias(self):
```

Comment:
Does not look like aliases to me 🤔 
Maybe we can come up with a better name?

I am also not sure what does this test do. Do we need to assert anything? Or it should just not crash?

Reply:
Originally this just didn't work in the initial typing extensions PR and the aliasing part of the PEP was removed, so good idea, I'll move this to test_basic

Reply:
Maybe we can also test that get_args() works right on these, something like `alias = Tuple[Self, Self]; assert get_args(alias) == (Self, Self)`.